### ISSUE TITLE
Update typescript-eslint to ^2.0.0

### ISSUE THREAD

**Issue by zakjan** [vuejs/eslint-config-typescript#2]
typescript-eslint 2.0.0 got released recently, which promises help with performance issues https://github.com/typescript-eslint/typescript-eslint/releases/tag/v2.0.0

**Comment by dacarley** [comment]
This would also remove warnings during vue-cli operations like:

```
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.2.1 <3.6.0

YOUR TYPESCRIPT VERSION: 3.6.3

Please only submit bug reports when using the officially supported version.

=============
```

**Comment by bcronje** [comment]
What changes do we have to apply to get the latest typescript-eslint working in vue-cli 4 ?

**Comment by zakjan** [comment]
Potentially can be closed by #3

**Comment by sodatea** [comment]
https://github.com/vuejs/eslint-config-typescript/releases/tag/v5.0.0
Released as v5.0.0
